src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonMigrator.cs

```diff
@@ -0,0 +1,51 @@
+﻿using NuGet.PackageManagement.UI;
+using NuGet.PackageManagement.VisualStudio;
+using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+
+namespace NuGet.VisualStudio
+{
+    [Export(typeof(IVsProjectJsonMigrator))]
+    class VsProjectJsonMigrator : IVsProjectJsonMigrator
+    {
+        private IVsSolutionManager _solutionManager;
+        private PumpingJTF _pumpingJtf;
+
+        [ImportingConstructor]
+        public VsProjectJsonMigrator()
+        {
+            _solutionManager = ServiceLocator.GetInstance<IVsSolutionManager>();
+            _pumpingJtf = new PumpingJTF(NuGetUIThreadHelper.JoinableTaskFactory.Context);
+        }
+        public IVsProjectJsonMigrateResult MigrateProjectToPackageRef(string projectUniqueName)
```

Comment:
This method should immediately call throw if not on the UI thread. Callers must call us from that so we can use DTE later.